src/games/strategy/triplea/ai/proAI/ProCombatMoveAI.java

```diff
@@ -612,13 +548,13 @@ private void removeTerritoriesThatArentWorthAttacking(final List<ProTerritory> p
         int minCost = Integer.MAX_VALUE;
         Unit minUnit = null;
         for (final Unit u : t.getUnits().getMatches(Matches.unitIsOwnedBy(player))) {
-          if (playerCostMap.getInt(u.getType()) < minCost) {
-            minCost = playerCostMap.getInt(u.getType());
+          if (ProData.unitValueMap.getInt(u.getType()) < minCost) {
```

Comment:
Logic in ProData would be my vote. You could at least unit test the full API. So even if the API gets large, at least it would work and you could refactor it easily down the road. In the meantime your client code is greatly simplified and you can get rid of some of the redundancy/complexity.